ext/bg/js/anki.js

```diff
@@ -19,122 +19,123 @@
     }
 
     async getModelNames() {
-        await this.checkVersion();
-        return await this.ankiInvoke('modelNames');
+        if (!this._enabled) { return []; }
+        await this._checkVersion();
+        return await this._invoke('modelNames');
     }
 
     async getModelFieldNames(modelName) {
-        await this.checkVersion();
-        return await this.ankiInvoke('modelFieldNames', {modelName});
+        if (!this._enabled) { return []; }
+        await this._checkVersion();
+        return await this._invoke('modelFieldNames', {modelName});
     }
 
     async guiBrowse(query) {
-        await this.checkVersion();
-        return await this.ankiInvoke('guiBrowse', {query});
+        if (!this._enabled) { return []; }
+        await this._checkVersion();
+        return await this._invoke('guiBrowse', {query});
     }
 
     async storeMediaFile(filename, dataBase64) {
-        await this.checkVersion();
-        return await this.ankiInvoke('storeMediaFile', {filename, data: dataBase64});
-    }
-
-    async checkVersion() {
-        if (this.remoteVersion < this.localVersion) {
-            this.remoteVersion = await this.ankiInvoke('version');
-            if (this.remoteVersion < this.localVersion) {
-                throw new Error('Extension and plugin versions incompatible');
-            }
+        if (!this._enabled) {
+            throw new Error('AnkiConnect not enabled');
         }
+        await this._checkVersion();
+        return await this._invoke('storeMediaFile', {filename, data: dataBase64});
     }
 
     async findNoteIds(notes) {
-        await this.checkVersion();
+        if (!this._enabled) { return []; }
+        await this._checkVersion();
         const actions = notes.map((note) => ({
             action: 'findNotes',
             params: {
-                query: `deck:"${AnkiConnect.escapeQuery(note.deckName)}" ${AnkiConnect.fieldsToQuery(note.fields)}`
+                query: `deck:"${this._escapeQuery(note.deckName)}" ${this._fieldsToQuery(note.fields)}`
             }
         }));
-        return await this.ankiInvoke('multi', {actions});
+        return await this._invoke('multi', {actions});
     }
 
-    ankiInvoke(action, params) {
-        return requestJson(this.server, 'POST', {action, params, version: this.localVersion});
+    // Private
+
+    async _checkVersion() {
+        if (this._remoteVersion < this._localVersion) {
+            this._remoteVersion = await this._invoke('version');
+            if (this._remoteVersion < this._localVersion) {
+                throw new Error('Extension and plugin versions incompatible');
+            }
+        }
+    }
+
+    async _invoke(action, params) {
+        const result = await requestJson(this._server, 'POST', {action, params, version: this._localVersion});
+        if (
+            result !== null &&
+            typeof result === 'object' &&
+            !Array.isArray(result)
+        ) {
```

Comment:
Can be replaced with `isObject(result)`

Reply:
018913d03fff627fb7d34f594340c47a607e8839